tests/kernel/context/src/main.c

```diff
@@ -245,6 +245,9 @@ static void _test_kernel_cpu_idle(int atomic)
 				k_cpu_idle();
 			}
 		} while ((idle_loops++ < CONFIG_MAX_IDLE_WAKES) && (idle_timer_done == false));
+		zassert_true(idle_timer_done,
+			     "The CPU was waken spuriously too many times (%d > %d)",
+			     idle_loops, CONFIG_MAX_IDLE_WAKES);
```

Comment:
The error message is incorrect, and is a result of the loop termination condition.

When CONFIG_MAX_IDLE_WAKES is 0, a single iteration `idle_loops` will always be 1. That is OK.

However, when CONFIG_MAX_IDLE_WAKES is 1, it looks to me that the loop will terminate after a single pass and the error message `(1 > 1)` will be incorrect.

Reply:
Thanks for the review @peter-mitsis 

> However, when CONFIG_MAX_IDLE_WAKES is 1, it looks to me that the loop will terminate after a single pass and the error message (1 > 1) will be incorrect.

If it exists without setting idle_time_done, it should have looped twice in that case (note the postincrement), so the msg should be "(2 > 1)"

Reply:
You are absolutely right. I got tripped up by the post-increment behaviour (again).